### PR TITLE
Add automatic labeler.

### PR DESCRIPTION
**Pull request by MaxJPRey**

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+- doc/source/**/*
+- README.md
+maintenance:
+- .github/**/*
+- .flake8
+- pyproject.toml
+- tox.ini
+- .pre-commit-config.yaml
+testing:
+- tests/*
```
Comment (AlejandroFernandezLuces) [inline .github/labeler.yml:11]:
```suggestion
- tests/*

```

File: .github/workflows/label.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    name: Set labels
+    needs: [label-syncer]
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+
+    # Label based on modified files
+    - name: Label based on changed files
```
Comment (AlejandroFernandezLuces) [inline .github/workflows/label.yml:34]: This comment is the same as the name, one of them should be removed. For the sake of consistency with line 40 I would avoid the name keyword in this step.

**Review by AlejandroFernandezLuces — APPROVED** [review]
LGTM, I added some minor comments.

File: .github/labeler.yml (hunk 1 above)
Comment (MaxJPRey) [reply .github/labeler.yml]: @AlejandroFernandezLuces For any reason, I thought I had apply your suggestion but it appears I dismissed it.
So, I will definitely add your suggestion.
Thanks @AlejandroFernandezLuces .

Comment (MaxJPRey) [inline .github/labeler.yml]:
```suggestion
- tests/*

```

File: .github/workflows/label.yml (hunk 2 above)
Comment (MaxJPRey) [reply .github/workflows/label.yml:34]: For those workflows, all PyANSYS use the same content (except the reference to a specific repo such as `pyansys/pyprimemesh`).

**Comment by MaxJPRey** [comment]
> @MaxJPRey Since all the labels were made to lower-case, could you correct the dependabot config file to add "maintenance" instead of "Maintenance" now?

@ninad-kamat  Of course, I will.

**Comment by MaxJPRey** [comment]
@ninad-kamat It should be `maintenance` everywhere.
Let me know if and where you keep seeing `Maintenance`.

**Comment by ninad-kamat** [comment]
> @ninad-kamat It should be `maintenance` everywhere. Let me know if and where you keep seeing `Maintenance`.

@MaxJPRey Please see labels in dependabot.yml in .github folder

**Comment by MaxJPRey** [comment]
> > @ninad-kamat It should be `maintenance` everywhere. Let me know if and where you keep seeing `Maintenance`.
> 
> @MaxJPRey Please see labels in dependabot.yml in .github folder

OK, got it. thanks @ninad-kamat . I though you were referring to the labels. yml.
Nice catch!

**Review by ninad-kamat — APPROVED** [review]

**Pull request merged**